ingenialink/canopen/network.py

```diff
@@ -119,34 +120,35 @@ class NetStatusListener(Thread):
 
     Args:
         network (CanopenNetwork): Network instance of the CANopen communication.
-        node (canopen.RemoteNode): Identifier for the targeted node ID.
 
     """
 
-    def __init__(self, network, node):
+    def __init__(self, network):
         super(NetStatusListener, self).__init__()
         self.__network = network
-        self.node = node
-        self.__timestamp = self.node.nmt.timestamp
-        self.__state = NET_STATE.CONNECTED
         self.__stop = False
 
     def run(self):
+        timestamps = {}
         while not self.__stop:
-            if self.__timestamp == self.node.nmt.timestamp:
-                if self.__state != NET_STATE.DISCONNECTED:
-                    self.__network.status = NET_STATE.DISCONNECTED
-                    self.__state = NET_STATE.DISCONNECTED
-                    self.__network._notify_status(NET_DEV_EVT.REMOVED)
-                else:
+            for node_id, node in list(self.__network._connection.nodes.items()):
```

Comment:
Why you use list instead of use items() output directly?

Reply:
This is what I mentioned in the daily, this way the node list is forced to reload.